src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -153,16 +153,6 @@ public static int Main(string[] args)
                 log.Error("Exception in RemoteTestAgent", ex);
             }
 
-            try
-            {
-                // Unregister the channel
-                testAgencyServer.Stop();
```

Comment:
@jnm2 Check my general comment and go back to the original issue. The services that are initialized in agent's Main() are the services the agent is supposed to have. It's not supposed to have any class derived from ServerBase, 'cause it's not a server. Due to an accident of how I implemented the agent (reusing the entire engine) the __code__ for the server is present, but we really don't want to start it just for it's side effects. We should instead figure out what a client actually needs and put it here - I suspect we will be putting it here __again__ if we do that.